invalid-name false-positive on Enum functional syntax on pylint v4

### Bug description

```python
"""Example"""
from enum import Enum
Color = Enum('Color', [('RED', 1), ('GREEN', 2), ('BLUE', 3)])
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
************* Module a
/tmp/a.py:3:0: C0103: Constant name "Color" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```

### Expected behavior

Since TypeVar is supported, I'd expect the functional syntax example from https://docs.python.org/3/library/enum.html to work

### Pylint version

```shell
pylint 4.0.0
astroid 4.0.1
Python 3.13.7 (main, Aug 14 2025, 00:00:00) [GCC 15.2.1 20250808 (Red Hat 15.2.1-1)]
```

--- a/doc/whatsnew/fragments/10660.false_positive
+++ b/doc/whatsnew/fragments/10660.false_positive
@@ -1,0 +1,4 @@
+Check enums created with the ``Enum()`` functional syntax to pass against the
+``--class-rgx`` for the :ref:`invalid-name` check, like other enums.
+
+Closes #10660

--- a/pylint/checkers/base/name_checker/checker.py
+++ b/pylint/checkers/base/name_checker/checker.py
@@ -14,7 +14,7 @@ import sys
 from collections.abc import Iterable
 from enum import Enum, auto
 from re import Pattern
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, cast
 
 import astroid
 from astroid import bases, nodes, util
@@ -484,12 +484,21 @@ class NameChecker(_BasicChecker):
                         )
                         return
 
-                # Check classes (TypeVar's are classes so they need to be excluded first)
-                elif isinstance(inferred_assign_type, nodes.ClassDef):
-                    self._check_name("class", node.name, node)
-
                 elif inferred_assign_type in (None, util.Uninferable):
                     return
+
+                # Check classes (TypeVar's are classes so they need to be excluded first)
+                elif isinstance(inferred_assign_type, nodes.ClassDef) or (
+                    isinstance(inferred_assign_type, bases.Instance)
+                    and "EnumMeta"
+                    in {
+                        ancestor.name
+                        for ancestor in cast(
+                            InferenceResult, inferred_assign_type
+                        ).mro()
+                    }
+                ):
+                    self._check_name("class", node.name, node)
 
                 # Don't emit if the name redefines an import in an ImportError except handler
                 # nor any other reassignment.

--- a/tests/functional/i/invalid/invalid_name.py
+++ b/tests/functional/i/invalid/invalid_name.py
@@ -117,3 +117,7 @@ class FooBar:
 class FooBarSubclass(FooBar):
     tearDown = FooBar.tearDown
     tearDownNotInAncestor = None  # [invalid-name]
+
+
+from enum import Enum
+Color = Enum('Color', [('RED', 1), ('GREEN', 2), ('BLUE', 3)])

